typescript/cli/src/utils/input.ts

```diff
@@ -77,6 +81,55 @@ export async function inputWithInfo({
   return answer;
 }
 
+/**
+ * Prompts the user to optionally set an existing ProxyAdmin contract address to be used in a WarpToken deployment.
+ */
+export async function setProxyAdminConfig(
+  context: CommandContext,
+  chain: ChainName,
+  warpRouteOwner: Address,
+): Promise<DeployedOwnableConfig> {
+  const defaultAdminConfig: DeployedOwnableConfig = {
+    owner: warpRouteOwner,
+  };
+
+  // default to deploying a new ProxyAdmin with `warpRouteOwner` as the owner
+  // if the user supplied the --yes flag
+  if (context.skipConfirmation) {
+    return defaultAdminConfig;
+  }
+
+  const useExistingProxy = await confirm({
+    message: `Use an existing Proxy Admin contract for the warp route deployment on chain "${chain}"?`,
+  });
+
+  if (!useExistingProxy) {
+    return defaultAdminConfig;
+  }
+
+  const proxyAdminAddress = await input({
+    message: `Please enter the address of the Proxy Admin contract to be used on chain "${chain}":`,
+    validate: isAddress,
+  });
+
+  const proxy = ProxyAdmin__factory.connect(
+    proxyAdminAddress,
+    context.multiProvider.getProvider(chain),
+  );
+
+  try {
+    const ownerAddress = await proxy.owner();
+    return {
+      address: proxyAdminAddress,
+      owner: ownerAddress,
+    };
+  } catch (error) {
+    throw new Error(
+      `Failed to read owner address from ProxyAdmin contract at ${proxy.address}. Are you sure this is a ProxyAdmin contract?`,
```

Comment:
nit: we can use any contract that is Ownable, but it doesn't mean that it's a ProxyAdmin. 

better to use other functions/ways to determine if it's a ProxyAdmin contract. but i understand that this may not be straight forward so i'm okay with this.